File: Plugins/Kingbotk/Branches/Version 1/AWB Plugin/AboutBox.vb
Namespace AutoWikiBrowser.Plugins.SDKSoftware.Kingbotk.Components
    Friend NotInheritable Class AboutBox
        Private Sub AboutBox_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Me.TextBoxDescription.Text = _
               "An AWB plugin for adding and updating WikiProject templates on Wikipedia talk pages. " & _
               Microsoft.VisualBasic.vbCrLf & Microsoft.VisualBasic.vbCrLf & "AWB Version: " & _
               Application.ProductVersion.ToString & Microsoft.VisualBasic.vbCrLf & Microsoft.VisualBasic.vbCrLf & _
               "Made in England. Store in a dry place and consume within 7 days of opening. COMES WITH NO WARRANTY - " _
               & "check your edits and use sensibly!"
        End Sub

        Private Sub OKButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles OKButton.Click
            Me.Close()
        End Sub

        Public Shared ReadOnly Property Version() As String
            Get
                Return System.Reflection.Assembly.GetExecutingAssembly.GetName.Version.ToString
            End Get
        End Property

        Public Sub New()
            ' This call is required by the Windows Form Designer.
            InitializeComponent()

            ' Add any initialization after the InitializeComponent() call.
            Me.LabelVersion.Text = String.Format("Version {0}", Version)
        End Sub
        Private Sub linkKingboy_LinkClicked(ByVal sender As System.Object, _
        ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles linkKingboy.LinkClicked
            linkKingboy.LinkVisited = True
            Tools.OpenENArticleInBrowser("Kingboyk", True)
        End Sub
        Private Sub linkReedy_LinkClicked(ByVal sender As System.Object, _
        ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles linkReedy.LinkClicked
            linkReedy.LinkVisited = True
            Tools.OpenENArticleInBrowser("Reedy Boy", True)
        End Sub
    End Class
End Namespace

File: tags/REL_4_3_2/AWB/AboutBox.cs
/*
Autowikibrowser
Copyright (C) 2007 Martin Richards
(C) 2007 Stephen Kennedy (Kingboyk) http://www.sdk-software.com/

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using System.Reflection;
using System.Threading;
using WikiFunctions;

namespace AutoWikiBrowser
{
    internal sealed partial class AboutBox : Form
    {
        public AboutBox() { } //default

        public AboutBox(string ieVersion, TimeSpan time, int Edits)
        {
            InitializeComponent();

            lblIEVersion.Text = "IE version: " + ieVersion;
            lblAWBVersion.Text = "AWB Version " + Program.VersionString;
            textBoxDescription.Text = WikiFunctions.Controls.AboutBox.GetDetailedMessage(Assembly.GetExecutingAssembly());
            lblOSVersion.Text = "Windows version: " + Environment.OSVersion.Version.Major.ToString() + "." + Environment.OSVersion.Version.Minor.ToString();
            lblNETVersion.Text = ".NET Version: " + Environment.Version.ToString();
            lblTimeAndEdits.Text = "You have made " + Edits.ToString() + " edits in " + time.ToString();
        }

        private void okButton_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void linkBluemoose_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkBluemoose.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Bluemoose", true);
        }

        private void linkLigulem_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkLigulem.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Ligulem", true);
        }

        private void linkMaxSem_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkMaxSem.LinkVisited = true;
            Tools.OpenENArticleInBrowser("MaxSem", true);
        }

        private void linkMets501_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkMets501.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Mets501", true);
        }

        private void linkReedy_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkReedy.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Reedy Boy", true);
        }

        private void linkKingboy_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkKingboy.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Kingboyk", true);
        }

        private void linkMartinp23_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkMartinp23.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Martinp23", true);
        }

        private void linkLabel2_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkAWBPage.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Wikipedia:AutoWikiBrowser", false);
        }
        
        private void linkLabel6_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkBugs.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Wikipedia_talk:AutoWikiBrowser/Bugs", false);
        }

        private void linkLabel7_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkFeatureRequests.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Wikipedia_talk:AutoWikiBrowser/Feature_requests", false);
        }

        private void linkJogers_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            linkJogers.LinkVisited = true;
            Tools.OpenENArticleInBrowser("Jogers", true);
        }

        private void UsageStatsLabel_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        { UsageStats.OpenUsageStatsURL(); }
    }
}
